### PR TITLE
opinfo : nn.functional.embedding

### PR DESCRIPTION
**Pull request by kshitij12345**
Adds opinfo for `nn.functional.embedding`

Few cases where `numerical` gradient doesn't match (gradcheck fails)

```python
import torch

try:
    t = torch.randn(2, 1, dtype=torch.float64, requires_grad=True)
    idx = torch.tensor([0, 1])
    torch.autograd.gradcheck(lambda idx, t : torch.nn.functional.embedding(idx, t, padding_idx=1), (idx, t, ))
except Exception as e:
    print("PADDING IDX:", e)

try:
    t = torch.ones(2, 1, dtype=torch.float64, requires_grad=True)
    idx = torch.tensor([0, 1])
    torch.autograd.gradcheck(lambda idx, t : torch.nn.functional.embedding(idx, t, max_norm=1.), (idx, t, ))
except Exception as e:
    print("MAX NORM:", e)

try:
    t = torch.randn(2, 1, dtype=torch.float64, requires_grad=True)
    idx = torch.tensor([0, 1, 1])
    torch.autograd.gradcheck(lambda idx, t : torch.nn.functional.embedding(idx, t, scale_grad_by_freq=True), (idx, t, ))
except Exception as e:
    print("SCALE GRAD BY FREQUENCY:", e)

try:
    t = torch.randn(2, 1, dtype=torch.float64, requires_grad=True)
    idx = torch.tensor([0, 1])
    torch.autograd.gradcheck(lambda idx, t : torch.nn.functional.embedding(idx, t, sparse=True), (idx, t, ))
except Exception as e:
    print("SPARSE", e)

```

### CODE REVIEW

**Review by zou3519 — APPROVED** [review]
LGTM! One really minor nit

**Comment by kshitij12345** [comment]
@soulitzer why was this reverted? Thanks

**Comment by soulitzer** [comment]
It didn't get reverted (again). I think this is just an issue with facebook-github-bot replaying the revert comment from earlier, so we probably don't need to worry. You can check that the revert commit has the same commit hash as in https://github.com/pytorch/pytorch/pull/66622#issuecomment-947826243.

**Comment by kshitij12345** [comment]
@soulitzer Ah, I see. Apologies for the noise. Thanks!
